src/mlpack/methods/ann/augmented/tasks/sort.hpp

```diff
@@ -0,0 +1,96 @@
+/**
+ * @file sort.hpp
+ * @author Konstantin Sidorov
+ *
+ * Definition of the SortTask class, which implements a generator of
+ * instances of sequence sort task.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_AUGMENTED_TASKS_SORT_HPP
+#define MLPACK_METHODS_AUGMENTED_TASKS_SORT_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /* Artificial Neural Network */ {
+namespace augmented /* Augmented neural network */ {
+namespace tasks /* Task utilities for augmented */ {
+
+/**
+ * Generator of instances of the sequence sort task.
+ * The parameters are:
+ * - maximum sequence length;
+ * - binary length of sequence elements.
+ * 
+ * Generated datasets are compliant with mlpack format -
+ * every dataset element is shaped as a vector of
+ * length (binary length) * (sequence length).
+ * 
+ * Example of generated dataset (sequence length = 3, binary length = 2):
+ * - Input sequences: [ [1,1,0,0,0,1], [0,0,0,0,1,1] ]
+ * - Output sequences: [ [0,0,0,1,1,1], [0,0,0,0,1,1] ]
```

Comment:
About the sequence representation: we use fixed-length representation (the generator first sample binary length of *all* sequence elements, then samples every sequence element as fixed-length binary vector). I have tried to make that more clear in the docs.